src/coreclr/jit/morph.cpp

```diff
@@ -353,7 +353,29 @@ GenTree* Compiler::fgMorphCast(GenTree* tree)
     }
     else if (((tree->gtFlags & GTF_UNSIGNED) == 0) && (srcType == TYP_LONG) && varTypeIsFloating(dstType))
     {
-        return fgMorphCastIntoHelper(tree, CORINFO_HELP_LNG2DBL, oper);
+        oper = fgMorphCastIntoHelper(tree, CORINFO_HELP_LNG2DBL, oper);
```

Comment:
Is this part of this change or some other fix?

Reply:
It is required.   CSE will assert because we can make a CSE with mismatched types